test/unit/controllers/declaration/CarrierDetailsControllerSpec.scala

```diff
@@ -75,30 +76,66 @@ class CarrierDetailsControllerSpec extends ControllerSpec {
 
   "Carrier Details Controller display page" should {
 
-    "return OK (200)" when {
+    onJourney(STANDARD, SIMPLIFIED, OCCASIONAL, CLEARANCE) { request =>
+      "return 200 (OK)" when {
 
-      onJourney(STANDARD, SIMPLIFIED, OCCASIONAL, CLEARANCE) { request =>
-        "with valid journey type" in {
+        "display page method is invoked and cache is empty" in {
 
-          val eori = Some(Eori("1234"))
-          withNewCaching(aDeclarationAfter(request.cacheModel, withCarrierDetails(eori)))
+          withNewCaching(request.cacheModel)
 
           val result = controller.displayPage(Mode.Normal)(getRequest())
 
-          status(result) mustBe OK
-          verifyPageInvocations(1)
+          status(result) must be(OK)
+        }
 
-          theResponseForm.value mustBe Some(CarrierDetails(EntityDetails(eori, None)))
-          theResponseForm.errors mustBe Seq.empty
+        "display page method is invoked and cache contains data" in {
+
+          withNewCaching(
+            aDeclarationAfter(
+              request.cacheModel,
+              withCarrierDetails(None, Some(Address("John Smith", "1 Export Street", "Leeds", "LS1 2PW", "United Kingdom")))
+            )
+          )
+
+          val result = controller.displayPage(Mode.Normal)(getRequest())
+
+          status(result) must be(OK)
         }
+      }
+    }
+
+    onJourney(STANDARD, SIMPLIFIED, OCCASIONAL, CLEARANCE) { request =>
+      "return 400 (BAD_REQUEST)" when {
+
+        "form is incorrect" in {
+
+          withNewCaching(request.cacheModel)
+
+          val incorrectForm = Json.toJson(CarrierDetails(EntityDetails(None, Some(Address("", "", "Leeds", "LS1 2PW", "United Kingdom")))))
 
+          val result = controller.saveAddress(Mode.Normal)(postRequest(incorrectForm))
+
+          status(result) must be(BAD_REQUEST)
+        }
       }
     }
 
     "return 303 (SEE_OTHER)" when {
 
-      "method is invoked and cache is empty" in {
+      onJourney(CLEARANCE) { request =>
+        "form is empty" in {
+
+          withNewCaching(request.cacheModel)
 
+          val incorrectForm = Json.toJson(CarrierDetails(EntityDetails(None, None)))
+
+          val result = controller.saveAddress(Mode.Normal)(postRequest(incorrectForm))
+
+          status(result) must be(SEE_OTHER)
+        }
+      }
+
+      "method is invoked and cache is empty" in {
```

Comment:
Should there be `onJourney` for all other journeys?

Reply:
Added for all other journey types